fdbserver/MoveKeys.actor.cpp

```diff
@@ -1590,14 +1590,24 @@ void seedShardServers(Arena& arena,
 			tr.set(arena, storageTeamIdKey(teamId), encodeStorageTeams(servers)); // TeamId -> Vec<StorageServers>
 			Key teamListKey = storageServerListToTeamIdKey(servers);
 			tr.set(arena, teamListKey, BinaryWriter::toValue(teamId, Unversioned())); // Vec<StorageServer> -> TeamId
-
 			if (servers.size() > 1) {
 				// Only seed team has more than one member.
 				ASSERT(!seedServerSet);
 				seedServerId = teamId;
 				seedServerSet = true;
 			}
 		}
+
+		if (!seedServerSet) {
+			ASSERT(serverToTeams.size() == 1);
```

Comment:
This assertion doesn't look right. We may have more than 1 server, and we may still have storage replication factor == 1 in the configuration.

Just remove this, and it should be ok.

Reply:
nice catch , fixed

Reply:
At this stage, where we are creating a new cluster, we only expect a seed team and private teams. Seed team will contain all the SS initially regardless what the RF is. This behaviour has been since the main branch :: https://github.com/apple/foundationdb/blob/main/fdbserver/MoveKeys.actor.cpp#L1549

I think once cluster is available, DD kicks in and builds teams that satisfy RF, but that should only happens after this.

Reply:
I think this case only happens if we have exactly one SS in the cluster, so we should assert for that.

Reply:
@halfprice please check comments above.

Reply:
Edit: `ASSERT(serverToTeams.size() == 1);` already does that.

Reply:
@vishesh this line is removed before this PR is merged, I am considering adding it back

Reply:
What if the replication factor is 1?

Reply:
Seed server team doesn't care about RF, even in main branch. E..g https://github.com/apple/foundationdb/blob/main/fdbserver/MoveKeys.actor.cpp#L1549, it will just create a team with entire key range with all servers which is `serverSrcUID`.

Reply:
The recruitment should follow the replication policy: https://github.com/apple/foundationdb/blob/1ab963eb5bd4e551966bfa8fef8c64ac31dd8b90/fdbserver/ClusterController.actor.h#L380.

If the replication policy is 1, the recruitment will only return 1 SS here, no matter how many storage processes we have in the cluster. Because of this, it makes sense to add back the assertion `ASSERT(serverToTeams.size() == 1);` here.

@hfu5 , can also also document this when you add it back?